src/passes/OptimizeInstructions.cpp

```diff
@@ -235,12 +235,18 @@ struct OptimizeInstructions
   // leave one of the two arbitrarily, so we might get unlucky).
   bool neverFold;
 
+  // As neverFold, but for reordering code. If we move a branch hint around code
+  // that might trap, and the trap happens later, the branch hint might start to
+  // execute, and it could be wrong.
+  bool neverReorder;
```

Comment:
Can we merge these two bools just like we merge the flag? They're always the same, right?

Reply:
Yeah, I just thought it reads a little simpler in each place in the code, `"if (!neverFold) fold()"` as opposed to `"if (!neverFoldOrSomethingElse) fold()"` - ?

Reply:
Fair enough 👍